### DISCUSSION TITLE
Does Selenium-Profiles support SeleniumWire?

### DISCUSSION THREAD

**Discussion by simplexx** [kaliiiiiiiiii/Selenium-Profiles#36] [Q&A] [answered]
Hello,

I am currently using Selenium Stealth, but get blocked by Cloudflare on many occasions, so I am looking for a better solution and this project looks very interesting. 

I use SeleniumWire to extract certain data from single requests, so I am wondering if I can use SeleniumWire together with Selenium-Profiles?

Cheers

**Comment by ttlns** [comment]
As I understand it, selenium-wire is only a proxy. Meaning there should be a way to start the selenium-wire proxy, and then specify with `profile["options"]["browser"]["proxy"]` and `profile["options"]["browser"]["proxy_method"]`

maybe you can start the seleniumwire-proxy with [backend.start](https://github.com/wkeeling/selenium-wire/blob/master/seleniumwire/backend.py)
Anyways, feel free to give me a feedback if you get it working:)

**Reply by simplexx** [reply]
Selenium-wire is a proxy, but it also adds new attributes to the webdriver object, such as driver.requests... and the driver is imported from the seleniumwire package like this: from seleniumwire import webdriver.
So as far as I understand, even if we add the proxy to the webdriver manually, these additional attributes would be missing in the webdriver object.
I thought that maybe selenium-profiles has a similar functionality to undetected-chromedriver, where selenium-wire can be used like this:
from seleniumwire import undetected_chromedriver - but it seems that's not the case unfortunately.

**Reply by ttlns** [reply]
Allright, see your point. Atm I don't have that much time for development, but this makes me consider implementing it🤔
 I might ask @wkeeling (owner of seleniumwire) in some time if he'd. 
be willing to add Selenium-Profiles : )

#### Update
@simplexx [seleniumwire/undetected_chromedriver/webdriver.py](https://github.com/wkeeling/selenium-wire/blob/master/seleniumwire/undetected_chromedriver/webdriver.py) seems to be the file in which selemium-wire gets wrapped around undetected-chromedriver. Pretty sure it's possible in a similar way for Selenium-Profiles. If you get it working bevore me, feel free to share the script, and I'll implement it.

**Comment by ttlns** [comment, marked as answer]
Seleniumwire is now supported.
See [Example-Script](https://github.com/kaliiiiiiiiii/Selenium-Profiles#modify-headers)

**Comment by ttlns** [comment]
Closing as it seems to be resolved
